### PR TITLE
[fix](outfile) Fix unable to export empty data

### LINKED ISSUE #30600: [Bug] Version 2.0.4 Unable to export empty data to hdfs / S3
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master、2.0.4

### What's Wrong?

 Unable to export empty data to hdfs / S3, this behavior is inconsistent with version 1.2.7, version 1.2.7 can export empty data to hdfs/ S3, and there will be exported files on S3/HDFS.

Here is the behavior on version 2.0.4:
1. If export empty data to s3, it will return success, but there are no exported files on S3.
Reproduction method:
```
select * from test_tb where id=0 INTO OUTFILE  "s3://repo/test-outfile-empty/my_file_"
properties (
    "AWS_ENDPOINT" = "xxx",
    "AWS_ACCESS_KEY" = "xxx",
    "AWS_SECRET_KEY"="xxx",
    "AWS_REGION" = "bj"
)
```
image

2. if export empty data to hdfs use broker, it will return failed because not open broker writer but close it.(if not use broker the behavior is same as export file to s3, exported files not exist)
Reproduction method:
```
select * from test_tb where id = 0 INTO OUTFILE "hdfs://host:port/data/export/empty/label_23_20_48_26_453655_1336993759/label_23_20_48_26_453748_1090390474"  PROPERTIES ( "broker.fs.defaultFS" = "hdfs://host:port","broker.name"="hdfs","broker.username"="user","broker.password"="password");
```
image

### What You Expected?

Behavior consistent with version 1.2.7

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zxealous**
## Proposed changes

Issue Number: close #30600
Fix unable to export empty data to hdfs / S3, this behavior is inconsistent with version 1.2.7, version 1.2.7 can export empty data to hdfs/ S3, and there will be exported files on S3/HDFS.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
run performance

**Comment by zxealous** [comment]
run buildall

File: be/src/io/fs/s3_file_writer.cpp [hunk 1]
```diff
@@ -202,17 +202,44 @@ Status S3FileWriter::close() {
...
-        if (_upload_id.empty()) {
+
+    if (_upload_id.empty()) {
+        if (_pending_buf != nullptr) {
+            // it might be one file less than 5MB, we do upload here
             auto* buf = dynamic_cast<UploadFileBuffer*>(_pending_buf.get());
             DCHECK(buf != nullptr);
             buf->set_upload_to_remote([this](UploadFileBuffer& b) { _put_object(b); });
+        } else {
+            // if there is no pending buffer, we need to create a empty file
```
Comment (morningman) [inline be/src/io/fs/s3_file_writer.cpp:213]:
```suggestion
            // if there is no pending buffer, we need to create an empty file
```

**Review by morningman — COMMENTED** [review]
LGTM, please add regression test case

**Comment by zxealous** [comment]
run p0

**Comment by zxealous** [comment]
run buildall

**Comment by zxealous** [comment]
run buildall

**Comment by zxealous** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by BePPPower — APPROVED** [review]
LGTM

**Comment by kaijchen** [comment]
Hi @ zxealous, thank you for contributing to Apache Doris.

Unfortunately, this PR caused a problem in the cloud pipeline, see #31148.
I haven't found the exact reason right now.
Please take a look if you have time.

**Pull request merged**